Source/Core/Core/PrimeHack/Mods/FpsControls.cpp

```diff
@@ -127,7 +149,162 @@ void FpsControls::run_mod_menu(Region region) {
   }
 }
 
+bool FpsControls::beam_visor_menu_handler(u32 cursor_base, u32 yaw_vel_address, Game game)
+{
+
+  if (CheckBeamMenuCtl() || CheckVisorMenuCtl())
+  {
+    handle_cursor(cursor_base + 0x9c, cursor_base + 0x15c, 0.95f, 0.90f);
+    write32(0, yaw_vel_address);
+
+    // Set the beam/visor menu state.
+    if (CheckBeamMenuCtl())
+    {
+      menu_cursor_state = Beam_Visor_Menu_State::BEAM_MENU_OPEN;
+    }
+    // implying that CheckVisorMenuCtl() == true instead;
+    else
+    {
+      menu_cursor_state = Beam_Visor_Menu_State::VISOR_MENU_OPEN;
+    }
```

Comment:
It's worth noting you should avoid being overly verbose. For instance: "Set the beam/visor menu state." 
This is already explicitly shown when you `menu_cursor_state =`

Comments are unnecessary if something is explicit, otherwise you're simply just stating it twice to the programmer. Instead save comments for unclear code where it is difficult to infer the information. Don't worry about having to explain C++ to us. Just note when you're doing something and it's not apparent as to why or if something needs mention of what it is (but keep it short).

I recommend shortening smaller logic like this, e.g moving the open brackets onto the same line, which cuts the line count down by nearly half:
```
if (CheckBeamMenuCtl()) {
   menu_cursor_state = Beam_Visor_Menu_State::BEAM_MENU_OPEN;
   } else { // implied that visor menu button pressed
   menu_cursor_state = Beam_Visor_Menu_State::VISOR_MENU_OPEN;
}
```
Or using ternary operators:
```
// If not beam then visor menu.
menu_cursor_state = CheckBeamMenuCtl() ? Beam_Visor_Menu_State::BEAM_MENU_OPEN : Beam_Visor_Menu_State::VISOR_MENU_OPEN;
```
This is only recommended if both results are implicitly clear and preferably the line isn't too long, so I would go with the former example in this scenario.